rclpy/src/rclpy/_rclpy.c

```diff
@@ -3436,28 +3436,14 @@ rclpy_take_raw_with_info(rcl_subscription_t * subscription, rmw_message_info_t *
 static PyObject *
 rclpy_message_info_to_dict(rmw_message_info_t * message_info)
 {
-  PyObject * dict = PyDict_New();
+  PyObject * dict = Py_BuildValue(
+    "{s:L,s:L}",
+    "source_timestamp", message_info->source_timestamp,
+    "received_timestamp", message_info->received_timestamp);
   if (dict == NULL) {
-    PyErr_Format(PyExc_RuntimeError, "Failed to create dictionary object");
+    PyErr_Format(PyExc_RuntimeError, "Failed to create dictionary object for message info");
```

Comment:
Do we still need this?  The [documentation](https://docs.python.org/3/c-api/arg.html#c.Py_BuildValue) for `Py_BuildValue` says:

```
 Returns the value or NULL in the case of an error; an exception will be raised if NULL is returned.
```

Reply:
You are right, there will be a SystemError exception set internal if NULL is returned.
`If there is an error in the format string, the SystemError exception is set and NULL returned.`

> Do we still need this?
No, we don't need it. But I think using detailed information raised by `RuntimeError` for this operation is more useful.

Reply:
All right, that's fair.